### PR TITLE
Add io.github.ardash.apple-flinger

### PR DESCRIPTION
**Pull request by faveoled**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.

Funny single- and multiplayer game - Use a slingshot to shoot with apples. Very challenging and addictive.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [ ] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [ ] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by faveoled** [comment]
bot, build

**Comment by faveoled** [comment]
bot, build io.github.ardash.apple-flinger

**Comment by faveoled** [comment]
Doesn't seem to be fixable without moving the repo

https://github.com/ar-/apple-flinger/

Flatpak doesn't allow such names:
error: 'io.github.ar-.apple-flinger' is not a valid application name: Only last name segment can contain -

File: io.github.ardash.apple-flinger.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+app-id: io.github.ardash.apple-flinger
+branch: '1.0'
```
Comment (hfiguiere) [inline io.github.ardash.apple-flinger.yaml:2]:
```suggestion
```

File: io.github.ardash.apple-flinger.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    
+      - type: file
+        url: https://gitlab.com/ar-/apple-flinger/-/raw/master/art/logo/logo.png?ref_type=heads&inline=false
+        dest-file: logo.png
+        sha256: d6aada75efe24a11f481654253d6b3b217ae0b6eae7badc908139b718386527d
+
+      - type: file
+        path: io.github.ardash.apple-flinger.desktop
+
+      - type: file
```
Comment (hfiguiere) [inline io.github.ardash.apple-flinger.yaml:42]: this one should be first in the list.

File: io.github.ardash.apple-flinger.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+modules:
+  - name: openjdk
+    buildsystem: simple
+    build-commands:
+    - /usr/lib/sdk/openjdk17/install.sh
+
+  - name: apple-flinger
+    buildsystem: simple
+    build-commands:
+      - cp apple-flinger.jar ${FLATPAK_DEST}/
```
Comment (hfiguiere) [inline io.github.ardash.apple-flinger.yaml:28]:
```suggestion
      - install -D apple-flinger.jar ${FLATPAK_DEST}/
```

File: io.github.ardash.apple-flinger.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+app-id: io.github.ardash.apple-flinger
+branch: '1.0'
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+command: run.sh
+rename-desktop: io.github.ardash.apple-flinger
```
Comment (hfiguiere) [inline io.github.ardash.apple-flinger.yaml:9]: this property doesn't exist

also it doesn't look like it has purpose.

File: io.github.ardash.apple-flinger.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+finish-args:
+  - --env=PATH=/app/jre/bin:/usr/bin:/app/bin
+  - --socket=fallback-x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline io.github.ardash.apple-flinger.yaml:13]: does this java app really support wayland?

File: io.github.ardash.apple-flinger.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    build-commands:
+    - /usr/lib/sdk/openjdk17/install.sh
+
+  - name: apple-flinger
+    buildsystem: simple
+    build-commands:
+      - cp apple-flinger.jar ${FLATPAK_DEST}/
+      - install -Dm644 logo.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      - install -Dm644 ${FLATPAK_ID}.desktop "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+      - install -D run.sh /app/bin/run.sh
```
Comment (hfiguiere) [inline io.github.ardash.apple-flinger.yaml:31]:
```suggestion
      - install -Dm755 run.sh /app/bin/run.sh
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]
required appstream file is missing.

**Comment by hfiguiere** [comment]
> Doesn't seem to be fixable without moving the repo
> 
> https://github.com/ar-/apple-flinger/
> 
> Flatpak doesn't allow such names:
> 
> ```
> error: 'io.github.ar-.apple-flinger' is not a valid application name: Only last name segment can contain -
> ```

the documentation say to replace `-` by `_`.

File: io.github.ardash.apple-flinger.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
+app-id: io.github.ardash.apple-flinger
```
Comment (hfiguiere) [inline io.github.ardash.apple-flinger.yaml:1]:
```suggestion
app-id: io.github.ar_.apple-flinger
```

Comment (hfiguiere) [reply io.github.ardash.apple-flinger.yaml:1]: files should be renamed, etc.

**Pull request closed without merging**